boards/xtensa/intel_adsp_cavs25/doc/intel_adsp_generic.rst

```diff
@@ -0,0 +1,338 @@
+(referred to as its "log port") and services requests on port 10000
+(its "requests port"). When you run West or Twister on your local host,
+it sends requests using the :zephyr_file:`soc/xtensa/intel_adsp/tools/cavstool_client.py`
+script (referred to as "CAVS tool client"). It also uses ports 9999 and 10000 on
+your local host, so be sure those ports are free.
+
+Flashing with West is simple.
+
+.. code-block:: console
+
+   west flash --remote-host remotehostname --pty remotehostname
+
+Running tests with Twister is slightly more complicated.
+
+.. code-block:: console
+
+   twister -p intel_adsp_cavs25 --device-testing --device-serial-pty="$ZEPHYR_BASE/soc/xtensa/intel_adsp/tools/cavstool_client.py,-s,remotehostname,-l" --west-flash="--remote-host=remotehostname" -T samples/hello_world
+
+If your network is set up such that the TCP connection from
+:file:`cavstool_client.py` to :file:`remote-fw-service.py` is forwarded through
+an intermediate host, you may need to tell :file:`cavstool_client.py` to connect
+to different ports as well as a different hostname. You can do this by appending
+the port numbers to the intermediate host name.
+
+.. code-block:: console
+
+   west flash --remote-host intermediatehost:reqport --pty remotehostname:logport
+   twister -p intel_adsp_cavs25 --device-testing --device-serial-pty="$ZEPHYR_BASE/soc/xtensa/intel_adsp/tools/cavstool_client.py,-s,remotehostname:logport,-l" --west-flash="--remote-host=remotehostname:reqport" -T samples/hello_world
+
+You can also save this information to a hardware map file and pass that to
+Twister.
+
+.. code-block:: console
+
+   twister -p intel_adsp_cavs25 --hardware-map cavs.map --device-testing -T samples/hello_world
+
+Here's a sample ``cavs.map``:
+
+.. code-block:: console
+
+   - connected: true
+     id: None
+     platform: intel_adsp_cavs25
+     product: None
+     runner: intel_adsp
+     serial_pty: "/home/zephyrus/zephyrproject/zephyr/soc/xtensa/intel_adsp/tools/cavstool_client.py,-s,remotehostname:logport,-l"
+     runner_params:
+     - --remote-host=remotehostname:reqport
+
+Any of the arguments you would pass to Twister or West, you can pass with the
+hardware map. As mentioned previously, you can see the Intel ADSP runner
+arguments by passing the ``--context`` flag while flashing with West.
+
+Refer to :ref:`twister_script` for more information on hardware maps.
+
+Local Flashing for West and Twister on CAVS
+-------------------------------------------
+
+You can also directly flash the signed binary with the CAVS tool on the board.
+This may be useful for debugging.
+
+.. code-block:: console
+
+   sudo ./cavstool.py zephyr.ri
+
+You should see the following at the end of the log if you are successful:
+
+.. code-block:: console
+
+   ***** Booting Zephyr OS vx.x.x-xxx-gxxxxxxxxxxxx *****
+   Hello World! intel_adsp_cavs25
+
+Flashing for ACE
+----------------
+
+Flashing is not yet supported for ACE boards, as ACE boards are not yet
```

Comment:
`ACE boards` is not quite right, there is nothing of that sort. `platfroms with the ACE based Audio DSP` is more appropriate.